actors/builtin/miner/miner_actor.go

```diff
@@ -397,49 +394,22 @@ func (a Actor) SubmitWindowedPoSt(rt Runtime, params *SubmitWindowedPoStParams)
 		sectorInfos, err := sectors.LoadForProof(postResult.Sectors, postResult.IgnoredSectors)
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load proven sector info")
 
-		// Skip verification if all sectors are faults.
-		// We still need to allow this call to succeed so the miner can declare a whole partition as skipped.
-		if len(sectorInfos) > 0 {
-			if len(params.Proofs) == 0 {
-				// The miner _was_ supposed to prove something, but didn't.
-				rt.Abortf(exitcode.ErrIllegalArgument, "no proofs submitted in window PoSt for %d sectors", len(sectorInfos))
-			}
-			// Verify the proof.
-			// A failed verification doesn't immediately cause a penalty; the miner can try again.
-			//
-			// This function aborts on failure.
-			verifyWindowedPost(rt, currDeadline.Challenge, sectorInfos, params.Proofs)
+		if len(sectorInfos) == 0 {
+			// Abort verification if all sectors are (now) faults.
```

Comment:
@anorth 
We could improve this saying that we abort verification when new faults (asks skipped faults) are larger than
Recovering + [PartitionSize - (AlreadyFaulty+Recovering)]/4.51.
Indeed, it is not rational to declare so many skipped faults. The miner loses less going for missing this deadline. 

CC @lucaniz @nicola

Reply:
I'd prefer to avoid anything complicated here – I think that policy logic belongs in the node, not the actors.

The only reason this was previously allowed was because there was a point when a skipped fault was cheaper than a detected fault, so we needed to support skipping everything. It's not just the rationality - there's nothing to prove. I think of this as just rejecting proof of nothing.

Reply:
>I'd prefer to avoid anything complicated here – I think that policy logic belongs in the node, not the actors.

Agreed, I opened an issues in the lotus repo (#4025)